govcd/ip_space_org_assignment.go

```diff
@@ -0,0 +1,160 @@
+	}
+	queryParams := queryParameterFilterAnd(fmt.Sprintf("ipSpaceRef.id==%s", ipSpace.IpSpace.ID), queryParameters)
+	typeResponses := []*types.IpSpaceOrgAssignment{{}}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryParams, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap all typeResponses into IpSpace types with client
+	results := make([]*IpSpaceOrgAssignment, len(typeResponses))
+	for sliceIndex := range typeResponses {
+		results[sliceIndex] = &IpSpaceOrgAssignment{
+			IpSpaceOrgAssignment: typeResponses[sliceIndex],
+			IpSpaceId:            ipSpace.IpSpace.ID,
+			vcdClient:            ipSpace.vcdClient,
+		}
+	}
+
+	return results, nil
+}
+
+// GetOrgAssignmentById retrieves IP Space Org Assignment with a given ID
+func (ipSpace *IpSpace) GetOrgAssignmentById(id string) (*IpSpaceOrgAssignment, error) {
+	if id == "" {
+		return nil, fmt.Errorf("IP Space Org Assignment lookup requires ID")
+	}
+
+	client := ipSpace.vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointIpSpaceOrgAssignments
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint, id)
+	if err != nil {
+		return nil, err
+	}
+
+	response := &IpSpaceOrgAssignment{
+		IpSpaceOrgAssignment: &types.IpSpaceOrgAssignment{},
+		IpSpaceId:            ipSpace.IpSpace.ID,
+		vcdClient:            ipSpace.vcdClient,
+	}
+
+	err = client.OpenApiGetItem(apiVersion, urlRef, nil, response.IpSpaceOrgAssignment, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// GetOrgAssignmentById retrieves IP Space Org Assignment with a given Org Name
+func (ipSpace *IpSpace) GetOrgAssignmentByOrgName(orgName string) (*IpSpaceOrgAssignment, error) {
+	if orgName == "" {
+		return nil, fmt.Errorf("name of Org is required")
+	}
+	queryParams := queryParameterFilterAnd(fmt.Sprintf("orgRef.name==%s", orgName), nil)
+	results, err := ipSpace.GetAllOrgAssignments(queryParams)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving IP Space Org Assignments by Org Name: %s", err)
+	}
+
+	singleResult, err := oneOrError("Org Name", orgName, results)
+	if err != nil {
+		return nil, err
+	}
+
+	return singleResult, nil
+}
+
+// GetOrgAssignmentById retrieves IP Space Org Assignment with a given Org ID
+func (ipSpace *IpSpace) GetOrgAssignmentByOrgId(orgId string) (*IpSpaceOrgAssignment, error) {
+	if orgId == "" {
+		return nil, fmt.Errorf("organization ID is required")
+	}
+	queryParams := queryParameterFilterAnd(fmt.Sprintf("orgRef.id==%s", orgId), nil)
+	results, err := ipSpace.GetAllOrgAssignments(queryParams)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving IP Space Org Assignments by Org ID: %s", err)
+	}
+
+	singleResult, err := oneOrError("Org ID", orgId, results)
+	if err != nil {
+		return nil, err
+	}
+
+	return singleResult, nil
+}
+
+// Update updates Org Assignment
```

Comment:
This reads strange :)

Reply:
It is probably not the first time.
The point is that starting Go's function comments with the function name itself is recommended. At least some documentation tools (like `go doc`) used to automatically inject it as the first word if it wasn't it.

And then it is me being overly simple to reuse the same verb :) I will see if I can rephrase it better.

Reply:
Maybe so?

`// Update Org Assignment`

Reply:
True, that is a simpler way without putting too many words. I have changed these occurrences.